### PR TITLE
CRM457-1662: Add a basic PDF renderer

### PR DESCRIPTION
**Pull request by patrick-laa**
## Description of change
Cribbing from CCQ, use Grover/Puppeteer to generate a PDF on demand and download it via a new button.

[Link to relevant ticket](https://dsdmoj.atlassian.net/browse/CRM457-1662)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "esbuild": "^0.21.5",
     "govuk-frontend": "5.4.0",
     "jquery": "^3.7.1",
+    "puppeteer": "^22.12.0",
```
Comment (jsugarman) [inline package.json:12]: is this being out of sync with the lib version a risk?

```
"puppeteer": "^19.2.0"
```

File: app/services/pdf_service.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      right: '1cm',
+    },
+    viewport: {
+      width: 2400,
+      height: 4800,
+    },
+    emulate_media: 'print',
+
+    launch_args: ['--font-render-hinting=medium', '--no-sandbox', '--force-renderer-accessibility'],
+  }.freeze
```
Comment (jsugarman) [inline app/services/pdf_service.rb:17]: could be extracted to an initializer but either way. [example from civil apply](https://github.com/ministryofjustice/laa-apply-for-legal-aid/blob/main/config/initializers/grover.rb)

File: .circleci/config.yml [hunk 3]
```diff
@@ -108,6 +108,13 @@ commands:
       - *install-assets
       - *save-assets
 
+  install-puppeteer:
+    steps:
+      - run:
+          name: Install Puppeteer with Chromium
+          command: |
+            yarn add puppeteer@19.2.0
```
Comment (jsugarman) [inline .circleci/config.yml:116]: do we need to lock it to 19.2.0. [example from civil apply](https://github.com/ministryofjustice/laa-apply-for-legal-aid/blob/62f68bd05d19d90472885c6e63621cc50b18b246/.circleci/config.yml#L196)

File: Dockerfile [hunk 4]
```diff
@@ -56,6 +56,25 @@ RUN mkdir -p /usr/src/app && \
...
+        ttf-freefont \
+        nodejs \
+        yarn
+
+# Tell Puppeteer to skip installing Chrome. We'll be using the installed package.
+ENV PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true
+ENV PUPPETEER_EXECUTABLE_PATH=/usr/bin/chromium-browser
+
+# Install latest version of Puppeteer that works with Chromium 108
+RUN yarn add puppeteer@19.2.0
```
Comment (jsugarman) [inline Dockerfile:76]: do we need to lock it to 19.2. [example from civl apply](https://github.com/ministryofjustice/laa-apply-for-legal-aid/blob/62f68bd05d19d90472885c6e63621cc50b18b246/docker/apply_base.dockerfile#L49)

**Review by jsugarman — APPROVED** [review]
looks fine to me, some suggestions cribbed from civil apply with example links

also, I see this [missing config is encouraged](https://pptr.dev/guides/configuration#changing-the-default-cache-directory), [example puppeteer.config.cjs from civil apply](https://github.com/ministryofjustice/laa-apply-for-legal-aid/blob/62f68bd05d19d90472885c6e63621cc50b18b246/puppeteer.config.cjs#L4)

File: package.json (hunk 1 above)
Comment (patrick-laa) [reply package.json:12]: Have removed the lib version

File: .circleci/config.yml (hunk 3 above)
Comment (patrick-laa) [reply .circleci/config.yml]: Have removed

File: app/services/pdf_service.rb (hunk 2 above)
Comment (patrick-laa) [reply app/services/pdf_service.rb:17]: Think I'm going to leave this as an option passed in for now, cos it makes debugging and tweaking easier - don't have to restart the local server to adjust the settings.

**Pull request merged**